### PR TITLE
Refactor CI to check and test code

### PR DESCRIPTION
**Pull request by F3Joule**
- Include `feature/*` and `staging` branches.
- Do not run if changed files don't include any of `.rs`, `.toml` or `.lock` files.

### CODE REVIEW

File: .github/workflows/test-code.yml [hunk 1]
```diff
@@ -1,12 +1,12 @@
-name: Test Code
+name: Check and test Code
 
 on:
   pull_request:
-    branches:
-      - main
+    branches: [ "main", "staging", "feature/*" ]
```
Comment (TarekkMA) [inline .github/workflows/test-code.yml:5]: It would be good to run on any PR

**Review by TarekkMA — CHANGES_REQUESTED** [review]

**Comment by F3Joule** [comment]
https://github.com/PureStake/moonbeam/blob/master/.github/workflows/build.yml

Use sccache as well to speed-up check and test processes as PureStake do,

File: .github/workflows/test-code.yml [hunk 2]
```diff
@@ -1,12 +1,9 @@
-name: Test Code
+name: Check and test Code
```
Comment (siman) [inline .github/workflows/test-code.yml:1]:
```suggestion
name: Check and Test Code
```

File: .github/workflows/test-code.yml [hunk 3]
```diff
@@ -45,7 +42,7 @@ jobs:
       #   run: cargo +${{ env.rustc_version }} clippy -- -D warnings
 
       - name: Check Code
-        run: cargo check
+        run: cargo check --release
 
       - name: Test Code
         run: cargo test
```
Comment (siman) [inline .github/workflows/test-code.yml:48]: Should it be also 
Check and Test Code
?

**Review by siman — CHANGES_REQUESTED** [review]

**Comment by TarekkMA** [comment]
This PR is also good place to check for code style/linting, after #77 is merged

File: .github/workflows/test-code.yml (hunk 2 above)
Comment (F3Joule) [reply .github/workflows/test-code.yml:1]: Why? :sweat_smile:

File: .github/workflows/test-code.yml (hunk 3 above)
Comment (F3Joule) [reply .github/workflows/test-code.yml:48]: No, two separate actions here.

**Comment by F3Joule** [comment]
> This PR is also good place to check for code style/linting, after #77 is merged

Yeah. Let's merge this PR first and then write this improvement to a backlog.

File: .github/workflows/test-code.yml [hunk 4]
```diff
@@ -45,7 +42,7 @@ jobs:
       #   run: cargo +${{ env.rustc_version }} clippy -- -D warnings
 
       - name: Check Code
-        run: cargo check
+        run: cargo check --release
 
       - name: Test Code
```
Comment (TarekkMA) [inline .github/workflows/test-code.yml:47]: Can you add another test to test all tests when benchmarking is enabled?

**Review by TarekkMA — APPROVED** [review]
Will add these improvemnts to backlog

**Review by siman — APPROVED** [review]

**Pull request merged**